config/config.go

```diff
@@ -8,56 +8,100 @@ import (
 	"errors"
 	"flag"
 	"fmt"
-	"io/ioutil"
+	"os"
 
 	"github.com/elastic/elastic-agent-client/v7/pkg/client"
 	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
 	"github.com/elastic/elastic-agent-libs/config"
 	"github.com/elastic/elastic-agent-libs/logp"
-	"github.com/elastic/go-ucfg/json"
 
 	"github.com/elastic/elastic-agent-shipper/monitoring"
 	"github.com/elastic/elastic-agent-shipper/output"
 	"github.com/elastic/elastic-agent-shipper/queue"
-	"github.com/elastic/elastic-agent-shipper/server"
+
+	"go.uber.org/zap/zapcore"
 )
 
 var (
-	configFilePath    string
+	configFilePath string
+	// ErrConfigIsNotSet reports that no unmanaged config file has been set
 	ErrConfigIsNotSet = errors.New("config file is not set")
+	fs                = flag.CommandLine
+	// Overwrites is the config map of CLI overwrites set by the -E flag
+	Overwrites = config.SettingFlag(fs, "E", "config overwrites")
+
+	esKey      = "elasticsearch"
+	consoleKey = "console"
+	kafaKey    = "kafka"
 )
 
 // A lot of the code here is the same as what's in elastic-agent, but it lives in an internal/ library
 func init() {
-	fs := flag.CommandLine
 	fs.StringVar(&configFilePath, "c", "", "Run the shipper in the unmanaged mode and use the given configuration file instead")
+
+}
+
+// ShipperClientConfig is the shipper-relevant portion of the config received from input units
+type ShipperClientConfig struct {
+	Server string     `config:"server" mapstructure:"server"`
+	TLS    ShipperTLS `config:"ssl" mapstructure:"ssl"`
+}
+
+// ShipperTLS is TLS-specific shipper client settings
+type ShipperTLS struct {
+	CAs  []string `config:"certificate_authorities" mapstructure:"certificate_authorities"`
+	Cert string   `config:"certificate" mapstructure:"certificate"`
+	Key  string   `config:"key" mapstructure:"key"`
+}
+
+// ShipperRootConfig defines the shipper config we get from elastic-agent's output unit
+type ShipperRootConfig struct {
+	Type    string        `config:"type"`
+	Shipper ShipperConfig `config:"shipper"`
 }
 
-// ShipperConfig defines the options present in the config file
+// ShipperConfig defines the config values stored under the `shipper` key in the fleet config
 type ShipperConfig struct {
-	Log     logp.Config       `config:"logging"`
+	// Don't know what logging config will look like via fleet yet,
```

Comment:
I don't think you need logging config. The shipper should log to stdout (or stderr) and the agent will write the logs to the elastic-agent-* logs and deal with log rotation for you. The log level to use comes in through the control protocol for each unit.